### DISCUSSION TITLE
Building a Behavior Classifier for Flies

### DISCUSSION THREAD

**Discussion by sahilsingh2402** [talmolab/sleap#1349] [General]
Hey everyone,

I have generated a pose dataset using Sleap that contains the positions of 9 flies for each frame. I want to build a behavior classifier based on this data, but I'm not sure how to approach the problem.

Sample: [https://drive.google.com/file/d/1W960Z92f1im80o1l6FveWXBQI5883iRx/view?usp=sharing](https://drive.google.com/file/d/1W960Z92f1im80o1l6FveWXBQI5883iRx/view?usp=sharing)

My goal is to create an input that takes 9 rows at once, where each row represents the position of one fly, and then learn from it by finding distances between each body part of the flies with each other to determine whether they are touching, grooming, or avoiding.

Additionally, I would like to consider past frames while predicting current frame outputs. Does anyone have suggestions on how to approach this problem? Are there any similar models or approaches that already exist for this?

I'm open to using various machine learning models such as decision trees, support vector machines, or even deep learning models.

If you have any insights or resources that could help me get started, please let me know! Thanks in advance.

**Comment by sronilsson** [comment]
Hi @sahilsingh2402 - When animals don’t shape shift too much (like mice), rather than distances, features that has helped me are intersections of lines, bounding boxes and points and their rolling aggregates, some basic use [here](https://github.com/sgoldenlab/simba/blob/master/docs/anchored_rois.md). You can use [shapely](https://github.com/shapely/shapely), says on the tin that it should be quick, I don’t always see it though.. I need tinkering or alternative for longer videos. Rolling aggregated for relevant [three-point angles](https://github.com/sgoldenlab/simba/blob/6f0067558bc85db79d1d841924189ff4aa693e06/simba/mixins/feature_extraction_mixin.py#L102), and maybe different combinations of key points for sub-animal [hull sizes](https://github.com/sgoldenlab/simba/blob/6f0067558bc85db79d1d841924189ff4aa693e06/simba/feature_extractors/perimeter_jit.py#L22) could help for behaviors you are interested in.
